### PR TITLE
Use autosummary for sphinx docs

### PR DESCRIPTION
**Pull request by coretl**
This looks nicer in general and is less error prone.

Also fix an issue in the tests where pre-commit wasn't being run as files weren't added to git yet

### CODE REVIEW

File: tests/test_example.py
Comment (callumforrester) [inline tests/test_example.py:1]: Should it have its own test rather than be part of a big one?

Comment (coretl) [reply tests/test_example.py:1]: yes, done

**Comment by coretl** [comment]
> Does this need an ADR?

Yes. Done

**Review by callumforrester — APPROVED** [review]
LGTM

**Pull request merged**